File: src/Actors/Pasajero.java
package Actors;

import EmployeeTasks.InterfazID;
import EmployeeTasks.InterfazPrioridad;
import java.util.Arrays;
import java.util.List;

public class Pasajero extends Usuario implements Comparable<Pasajero>, InterfazPrioridad, InterfazID{
    
    private String prioridad;
    private int numeroAsiento;
    
    public Pasajero(String nombre, String apellidos, String telefono, String identificacion, String correo, String contraseña, String rol, String prioridad, int numeroAsiento) {
        super(nombre, apellidos, telefono, identificacion, correo, contraseña, rol);
        this.numeroAsiento = numeroAsiento;
        this.prioridad = prioridad;
    }

    @Override
    public String getPrioridad() {
        return prioridad;
    }

    public void setPrioridad(String prioridad) {
        this.prioridad = prioridad;
    }

    public int getNumeroAsiento() {
        return numeroAsiento;
    }

    public void setNumeroAsiento(int numeroAsiento) {
        this.numeroAsiento = numeroAsiento;
    }

    @Override
    public String getId() {
        return getIdentificacion();
    }

    @Override
    public int compareTo(Pasajero otro) {
        List<String> ordenPrioridades = Arrays.asList("premium", "ejecutivo", "estándar");

        int thisIndex = ordenPrioridades.indexOf(this.prioridad.toLowerCase());
        int otroIndex = ordenPrioridades.indexOf(otro.prioridad.toLowerCase());

        return Integer.compare(thisIndex, otroIndex);
    }

    @Override
    public String toString() {
        return "Pasajero{" + "prioridad=" + prioridad + ", numeroAsiento=" + numeroAsiento + '}';
    }

File: Códigos/Actores/Pasajero.cs
using ProyectoEstructuras.Códigos.TareasEmpleado;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


namespace ProyectoEstructuras.Códigos.Actores
{
    public class Pasajero: Usuario, IComparable<Pasajero>, InterfazPrioridad, InterfazID

    {
        private string prioridad;
        private int numeroAsiento;

        public string Prioridad { get => prioridad; set => prioridad = value; }
        public int NumeroAsiento { get => numeroAsiento; set => numeroAsiento = value; }

        public string GetPrioridad() { return Prioridad; }
        public string GetId() { return Identificacion; }

        public Pasajero(string nombre, string apellidos, string telefono, string identificacion, string correo, string contrasenia, string rol, string prioridad, int numeroAsiento)
            : base(nombre, apellidos, telefono, identificacion, correo, contrasenia, rol)
        {
            this.Prioridad = prioridad;
            this.NumeroAsiento = numeroAsiento;
        }


        public int CompareTo(Pasajero otro)
        {
            List<string> ordenPrioridades = new List<string> { "premium", "ejecutivo", "estándar" };

            int thisIndex = ordenPrioridades.IndexOf(this.Prioridad.ToLower());
            int otroIndex = ordenPrioridades.IndexOf(otro.Prioridad.ToLower());

            return thisIndex.CompareTo(otroIndex);
        }

        public override string ToString()
        {
            return $"Pasajero{{prioridad={Prioridad}, numeroAsiento={NumeroAsiento}}}";
        }

    }
}
